examples/ode.py

```diff
@@ -0,0 +1,112 @@
+"""
+
+import argparse
+import os
+
+import matplotlib
+import matplotlib.pyplot as plt
+
+from jax.experimental.ode import build_odeint
+import jax.numpy as np
+from jax.random import PRNGKey
+
+import numpyro
+import numpyro.distributions as dist
+from numpyro.examples.datasets import LYNXHARE, load_dataset
+from numpyro.infer import MCMC, NUTS, Predictive
+
+matplotlib.use('Agg')  # noqa: E402
+
+
+def dz_dt(z, t, alpha, beta, gamma, delta):
+    """
+    Lotka–Volterra equations. Real positive parameters `alpha`, `beta`, `gamma`, `delta`
+    describes the interaction of two species.
+    """
+    u = z[0]
+    v = z[1]
+    du_dt = (alpha - beta * v) * u
+    dv_dt = (-gamma + delta * u) * v
+    return np.stack([du_dt, dv_dt])
+
+
+predator_prey_int = build_odeint(dz_dt, rtol=1e-5, atol=1e-3, mxstep=500)
+
+
+def model(N, y=None):
+    """
+    :param int N: number of measurement times
+    :param numpy.ndarray y: measured populations with shape (N, 2)
+    """
+    # initial population
+    z_init = numpyro.sample("z_init", dist.LogNormal(np.log(10), 1), sample_shape=(2,))
+    # measurement times
+    ts = np.arange(float(N))
+    # parameters alpha, beta, gamma, delta of dz_dt
+    theta = numpyro.sample(
+        "theta",
+        dist.TruncatedNormal(low=0., loc=np.array([0.5, 0.05, 1.5, 0.05]),
+                             scale=np.array([0.5, 0.05, 0.5, 0.05])))
+    # integrate dz/dt, the result will have shape N x 2
+    z = predator_prey_int(z_init, ts, *theta)
+    # measurement errors, we expect that measured hare has larger error than measured lynx
+    sigma = numpyro.sample("sigma", dist.Exponential(np.array([1, 2])))
+    # measured populations (in log scale)
+    numpyro.sample("y", dist.Normal(np.log(z), sigma), obs=y)
+
+
+def main(args):
+    _, fetch = load_dataset(LYNXHARE, shuffle=False)
+    year, data = fetch()  # data is in hare -> lynx order
+
+    # use dense_mass for better mixing rate
+    mcmc = MCMC(NUTS(model, dense_mass=True),
+                args.num_warmup, args.num_samples, num_chains=args.num_chains,
+                progress_bar=False if "NUMPYRO_SPHINXBUILD" in os.environ else True)
+    mcmc.run(PRNGKey(1), N=data.shape[0], y=np.log(data))
+    mcmc.print_summary()
+
+    # predict populations
+    y_pred = Predictive(model, mcmc.get_samples())(PRNGKey(2), data.shape[0])["y"]
+    pop_pred = np.exp(y_pred)
+    plt.fill_between(year, data[:, 0], color="y", alpha=0.2, label="true hare")
+    plt.plot(year, data[:, 1], "kx", label="true lynx")
+    plt.plot(year, np.mean(pop_pred[..., 0], 0), "b-", label="pred hare")
+    plt.plot(year, np.mean(pop_pred[..., 1], 0), "g--", label="pred lynx")
+    plt.gca().set(ylim=(0, 160), xlabel="year", ylabel="population (in thousands)")
+    handles, labels = plt.gca().get_legend_handles_labels()
+    order = [3, 0, 1, 2]
+    plt.legend([handles[idx] for idx in order], [labels[idx] for idx in order])
+
+    plt.savefig("ode_plot.pdf")
```

Comment:
Also add this example and the figure to the gallery?

Reply:
It will be added automatically. We have filtered only a few examples: https://github.com/pyro-ppl/numpyro/blob/master/notebooks/source/conf.py#L180

Reply:
Oh that's great, I didn't realize that.

Reply:
The downside is when building the website, if the content of a script is changed, it will rerun the script to capture the output. I guess it is fine given than we only need to update the examples when we pump up the version (actually, it will be a good time to verify that we still get correct outputs). :)